lisa/util/__init__.py

```diff
@@ -466,18 +464,9 @@ def find_group_in_lines(
 
 
 def deep_update_dict(src: Dict[str, Any], dest: Dict[str, Any]) -> Dict[str, Any]:
-    if (
-        dest is None
-        or isinstance(dest, int)
-        or isinstance(dest, bool)
-        or isinstance(dest, float)
-        or isinstance(dest, str)
-    ):
-        result = dest
-    else:
-        result = dest.copy()
 
-    if isinstance(result, dict):
+    if isinstance(dest, dict):
```

Comment:
This change may have regression. If the dest is int, bool, float or str, the changed logic will take src, instead of dest.

Reply:
The behavior hasn't changed. In the old logic, if `dest` was an `int`, `bool`, `float` or `str`, then `result` would not be a `dict`, causing `result` to fall back to `src`. This was all just useless logic.